gateway/modules/auth/match.go

```diff
@@ -72,8 +74,48 @@ func (m *Module) matchRule(ctx context.Context, project string, rule *config.Rul
 }
 
 func (m *Module) matchFunc(ctx context.Context, rule *config.Rule, MakeHTTPRequest utils.TypeMakeHTTPRequest, args map[string]interface{}) error {
-	obj := args["args"].(map[string]interface{})
-	token := obj["token"].(string)
+	newArgs := args["args"].(map[string]interface{})
+
+	var token string
+	var claims interface{}
+	var err error
+	if len(rule.Claims) > 0 {
+		token, err = m.CreateToken(ctx, rule.Claims)
+		if err != nil {
+			return formatError(ctx, rule, helpers.Logger.LogError(helpers.GetRequestID(ctx), "Unable to create new token used by the webhook url in security rule (Webhook)", err, nil))
+		}
+		claims = rule.Claims
+		newArgs["auth"] = rule.Claims
+	} else {
+		claims = newArgs["auth"]
+		token = newArgs["token"].(string)
+	}
+
+	var obj interface{}
+	if rule.ReqTmpl != "" {
+		switch rule.Template {
+		case config.TemplatingEngineGo:
+			// Create a new template object
+			t := template.New(rule.Name)
+			t = t.Funcs(tmpl2.CreateGoFuncMaps(m))
+			t, err = t.Parse(rule.ReqTmpl)
+			if err != nil {
+				return formatError(ctx, rule, helpers.Logger.LogError(helpers.GetRequestID(ctx), "Unable to parse provided template in security rule (Webhook)", err, nil))
+			}
+			if rule.OpFormat == "" {
+				rule.OpFormat = "json"
+			}
+			obj, err = tmpl2.GoTemplate(ctx, t, rule.OpFormat, token, claims, args["args"])
```

Comment:
put the original token and claims here

Reply:
If I pass original token here, then what is the use of creating new token & claims if we are not using it?

Reply:
The point of claims is to generate a custom token to override the original token in the Authorisation header. This is useful when the webhook is on a SC endpoint and we want to make sure the request was made internally.